scene/3d/visibility_notifier.cpp

```diff
@@ -42,6 +42,10 @@ void VisibilityNotifier::_enter_camera(Camera *p_camera) {
 	ERR_FAIL_COND(cameras.has(p_camera));
 	cameras.insert(p_camera);
 
+	if (is_inside_tree() && Engine::get_singleton()->is_editor_hint()) {
+		return;
+	}
+
```

Comment:
These can be wrapped in `#ifdef TOOLS_ENABLED` I think, as this can be compiled out in release non-tools builds.

Reply:
Makes sense. I'll update accordingly.